plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection-tester.block_theme.spec.ts

```diff
@@ -357,3 +357,111 @@ test.describe( 'Product Collection registration', () => {
 		} );
 	} );
 } );
+
+test.describe( 'Deleted product reference in Product Collection block', () => {
+	let testProductId: number | null = null;
+
+	test.beforeEach( async ( { requestUtils } ) => {
+		// Activate plugin
+		await requestUtils.activatePlugin(
+			'register-product-collection-tester'
+		);
+
+		// Check if "A Test Product" exists
+		const products = await requestUtils.rest( {
+			method: 'GET',
+			path: 'wc/v3/products',
+			params: {
+				search: 'A Test Product',
+			},
+		} );
+
+		// Add "Test Product" only if it doesn't exist
+		if ( products.length === 0 ) {
+			const newProduct = await requestUtils.rest( {
+				method: 'POST',
+				path: 'wc/v3/products',
+				data: {
+					name: 'A Test Product',
+					price: 10,
+				},
+			} );
+			testProductId = newProduct.id;
+		} else {
+			testProductId = products[ 0 ].id;
+		}
+	} );
```

Comment:
Hmmm, this should not be necessary, as we always start each test with the same DB state. Is it somehow conditionally added with the plugin?